libscript/include/pivot/script/Stack.hxx

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include "pivot/ecs/Core/Data/value.hxx"
+
+#include <functional>
+#include <unordered_map>
+
+namespace pivot::ecs::script::interpreter
+{
+
+struct Variable {
+    std::string name;
+    bool hasValue = false;
+    data::Value value;
+    std::unordered_map<std::string, Variable> members;
+};
+
+/**
+ * \brief	Container for all the variables a system
+ * 			holds during the execution of a tick
+ *
+ * 	This container allows access, use and modification of the
+ * 	variables it stores. This is use by the interpreter to handle
+ * 	variables during the execution of a tick.
+ */
+class Stack
```

Comment:
Could be a template class ?
```cpp
template<typename Key, typename Value>
```

Reply:
Usage is pretty specific, and requires recursion in the Value field (since it searches that way inside the container). Therefore, I don't think there's enough other use cases for it to be templated.